modules/common/cronjob/cronjob.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2022 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cronjob
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	batchv1 "k8s.io/api/batch/v1"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// NewCronJob returns an initialized CronJob.
+func NewCronJob(
+	cronjob *batchv1.CronJob,
+	timeout int,
```

Comment:
As a generic rule I suggest to always use time.Duration for timeout parameters. This allows the caller to specify a lot more fine grained timeout value. 
One reason is that in envtest, running locally, I can use sub second timeout values to speed up the test execution.
Another good thing about time.Duration that it encodes the unit of the timeout value in the type itself.

Reply:
Sounds reasonable. I pushed one more commit to convert usage of int in all modules.

Reply:
@kajinamit I appreciate your eagerness to fix it all over the place but please note that this is a breaking change in other operators that already using the interface base on the int type. So I suggest to fix it for CronJob here as it is not used yet. And do the overall cleanup separately by not just fixing the lib-common interface but also adjusting the operator codes that are using it.

Reply:
I agree with the concern but I'm more concerned with creating inconsistent interfaces and ended up leaving it eventually. That would cause confusion for long term maintenance.

I can update the other modules to adapt to this interface change, but if now is not a good timing to make this change in all operators then I prefer using it consistently across all and then replace all by time.Duration at the same time.

Reply:
I agree with the plan. Updated the change to touch only cronjob for now. Will create a separate PR(s) to update the other modules.

Reply:
Cool, thanks. Add me to the new PR and will try to help push it through